libs/application/data-providers/src/providers/NationalRegistryProvider.ts

```diff
@@ -32,31 +33,25 @@ export class NationalRegistryProvider extends BasicDataProvider {
       }
     `
 
-    return this.useGraphqlGateway(query)
-      .then(async (res: Response) => {
-        const response = await res.json()
-        if (response.errors) {
-          return this.handleError(response.errors)
-        }
-
-        return Promise.resolve(response.data.nationalRegistryUser)
-      })
-      .catch((error) => {
-        return this.handleError(error)
-      })
-  }
+    return this.useGraphqlGateway(query).then(async (res: Response) => {
+      const response = await res.json()
+      if (response.errors) {
+        console.error(
+          `graphql error in ${this.type}: ${response.errors[0].message}`,
+        )
+        return Promise.reject({})
+      }
 
-  handleError(error: any) {
-    console.log('Provider error - NationalRegistry:', error)
-    return Promise.resolve({})
+      return Promise.resolve(response.data.nationalRegistryUser)
+    })
   }
 
-  onProvideError(result: string): FailedDataProviderResult {
+  onProvideError(result: unknown): FailedDataProviderResult {
     return {
       date: new Date(),
-      reason: result,
+      reason: coreErrorMessages.errorDataProvider,
```

Comment:
Would it not be good that this message indicates where it went wrong? This will just say that something went wrong fetching your data, not that it is connected to the national registry.

Reply:
good point.. it does show up below the national registry provider:

![image](https://user-images.githubusercontent.com/61151/132861911-18d08036-f149-4a9c-a00b-c847459b2d31.png)

Perhaps I should just add inline translation strings or at least along with the provider - do we have a translation namespace for global providers such as this one?